src/dune_async_io/async_io.ml

```diff
@@ -224,7 +224,11 @@ let start t =
   let module Scheduler = (val t.scheduler : Scheduler) in
   Scheduler.spawn_thread (fun () ->
     Mutex.lock t.mutex;
-    Exn.protect ~f:(fun () -> select_loop t) ~finally:(fun () -> Mutex.unlock t.mutex))
+    Exn.protect
+      ~f:(fun () -> select_loop t)
+      ~finally:(fun () ->
+        try Mutex.unlock t.mutex with
```

Comment:
Let's add a comment explaining why we are catching the error here.

Reply:
Something about this code didn't sit right with me and I rewrote it without this ugliness. Take another look.